zubhub_frontend/zubhub/src/views/create_activity/CreateEditActivity.jsx

```diff
@@ -102,45 +104,40 @@ export default function CreateActivity(props) {
   const formikStep2 = useFormik(script.step2Schema);
 
   const previous = () => go('prev');
-  const next = async () => {
-    let error = await checkErrors();
 
-    console.log(error);
-    if (Object.keys(error || {}).length > 0) return;
+  const submit = async publish => {
+    const errors = await checkErrors().then(errors => errors);
+
+    if (Object.keys(errors).length > 0) {
+      Object.keys(errors).map(err => {
+        return toast.error(err);
+      });
+      return;
+    }
 
     setIsLoading(true);
 
-    script.submitForm(
-      {
-        step1Values: formikStep1.values,
-        step2Values: formikStep2.values,
-        props: { ...props, auth },
-        state,
-        handleSetState: handleSetState,
-        step: activeStep,
-      },
-      success => {
-        if (success) {
-          if (activeStep == 1) go('next');
-          if (activeStep == 2) props.history.push(`/activities/${props.match.params.id}?success=true`);
-          setIsLoading(false);
-        }
-      },
-    );
+    script.submitForm({
+      step1Values: formikStep1.values,
+      step2Values: formikStep2.values,
+      props: { ...props, auth },
+      state,
+      handleSetState: handleSetState,
+      step: activeStep,
+      publish: publish,
+    });
   };
 
   const checkErrors = () => {
-    if (activeStep === 1) {
-      return formikStep1.setTouched({ title: true }, true);
-    }
-
-    if (activeStep === 2) {
-      return formikStep2.setTouched({ introduction: true }, true);
-    }
+    return formikStep2.setTouched({ introduction: true }, true);
   };
 
-  const go = direction => {
+  const go = async direction => {
```

Comment:
refactored the `go` function such that it first checks for validation errors before going to the next step